### PR TITLE
chore: put server side filtering related config under feature flag for table widget

### LINKED ISSUE #25530: [Task]: Enable feature flag for server side filtering table widget
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

The task here is to add the server side filtering feature of table widget under feature flag.

### PR DESCRIPTION
**Pull request by keyurparalkar**
## Description
This PR puts the `server side filtering` and `onTableFilterUpdate` properties in the property pane under the feature flag. It also puts the `Table1.filters` autocomplete inside a feature flag.

#### PR fixes following issue(s)
Fixes #25530

#### Type of change
- New feature (non-breaking change which adds functionality)

#### How Has This Been Tested?

- [ ] Manual
- [ ] Jest
- [ ] Cypress
    - Added cypress test to check that `.filters` and property pane configs are not visible when feature flag is enabled.
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/widgets/TableWidgetV2/widget/propertyConfig/contentConfig.ts [hunk 1]
```diff
@@ -237,6 +238,8 @@ export default [
         isBindProperty: false,
         isTriggerProperty: false,
         defaultValue: false,
+        hidden: () =>
+          !Widget.getFeatureFlag("release_table_serverside_filtering_enabled"),
```
Comment (sbalaji1192) [inline app/client/src/widgets/TableWidgetV2/widget/propertyConfig/contentConfig.ts:242]: `ALLOW_TABLE_WIDGET_SERVER_SIDE_FILTERING`

File: app/client/src/widgets/TableWidgetV2/widget/index.tsx [hunk 2]
```diff
@@ -303,6 +305,14 @@ class TableWidgetV2 extends BaseWidget<TableWidgetProps, WidgetState> {
 
   static getAutocompleteDefinitions(): AutocompletionDefinitions {
     return (widget: TableWidgetProps, extraDefsToDefine?: ExtraDef) => {
+      const featureFlagBasedAutoCompleteConfig: Record<string, any> = {};
+
+      if (this.releaseSeverSideFilterFeatureFlag) {
```
Comment (sbalaji1192) [inline app/client/src/widgets/TableWidgetV2/widget/index.tsx:310]: no need to define a class variable `releaseSeverSideFilterFeatureFlag`.

File: app/client/src/widgets/TableWidgetV2/widget/index.tsx [hunk 3]
```diff
@@ -303,6 +305,14 @@ class TableWidgetV2 extends BaseWidget<TableWidgetProps, WidgetState> {
 
   static getAutocompleteDefinitions(): AutocompletionDefinitions {
     return (widget: TableWidgetProps, extraDefsToDefine?: ExtraDef) => {
+      const featureFlagBasedAutoCompleteConfig: Record<string, any> = {};
```
Comment (sbalaji1192) [inline app/client/src/widgets/TableWidgetV2/widget/index.tsx:308]: use proper type

File: app/client/src/widgets/TableWidgetV2/widget/index.tsx [hunk 4]
```diff
@@ -332,9 +342,8 @@ class TableWidgetV2 extends BaseWidget<TableWidgetProps, WidgetState> {
         isAddRowInProgress: "bool",
         previousPageVisited: generateTypeDef(widget.previousPageVisited),
         nextPageVisited: generateTypeDef(widget.nextPageButtonClicked),
-        filters: generateTypeDef(widget.filters),
+        ...featureFlagBasedAutoCompleteConfig,
       };
```
Comment (sbalaji1192) [inline app/client/src/widgets/TableWidgetV2/widget/index.tsx:346]: this won't be very clear in the future. You should add the condition `if (this.releaseSeverSideFilterFeatureFlag) {` after you define the config object. no need to create a variable, mutate the config object directly

**Comment by laveena-en** [comment]
@keyurparalkar Folks with moolya accounts cant seem to access the feature. Can you please check?

**Comment by keyurparalkar** [comment]
@laveena-en Can you now try to access the dp? I just created the account with moolya domain on DP and its working for me.

**Comment by laveena-en** [comment]
Verified. Working now! @keyurparalkar

**Comment by keyurparalkar** [comment]
@laveena-en can we do sanity checks on test cases mentioned in this PR: https://github.com/appsmithorg/appsmith/pull/25732?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/TableV2_Widget_Copy_Paste_spec.js [hunk 5]
```diff
@@ -42,7 +42,11 @@ describe("Test Suite to validate copy/paste table Widget V2", function () {
     });
     cy.wait(200);
     cy.get(apiwidget.propertyList).then(function ($lis) {
-      expect($lis).to.have.length(23);
+      /*
```
Comment (sbalaji1192) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/TableV2_Widget_Copy_Paste_spec.js:45]: ideally we should test that it turns to 23 if we turn on server sider filtering flag

File: app/client/src/widgets/TableWidgetV2/widget/index.tsx [hunk 6]
```diff
@@ -332,9 +333,12 @@ class TableWidgetV2 extends BaseWidget<TableWidgetProps, WidgetState> {
         isAddRowInProgress: "bool",
         previousPageVisited: generateTypeDef(widget.previousPageVisited),
         nextPageVisited: generateTypeDef(widget.nextPageButtonClicked),
-        filters: generateTypeDef(widget.filters),
       };
 
+      if (BaseWidget.getFeatureFlag(ALLOW_TABLE_WIDGET_SERVER_SIDE_FILTERING)) {
```
Comment (sbalaji1192) [inline app/client/src/widgets/TableWidgetV2/widget/index.tsx:338]: you can use widget here instead of BaseWidget

**Review by sbalaji1192 — APPROVED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/server_side_filtering_spec_1.ts [hunk 7]
```diff
@@ -113,3 +110,21 @@ describe("Table widget v2: test server side filtering", function () {
...
+  before(() => {
+    featureFlagIntercept({ release_table_serverside_filtering_enabled: false });
+    entityExplorer.DragDropWidgetNVerify(draggableWidgets.TABLE, 700, 300);
+  });
+  it("1. should test that server side filtering option should not be visible in the property pane", () => {
+    agHelper.AssertElementAbsence(
+      propPane._propertyControl("serversidefiltering"),
+    );
+  });
+  it("2. should test that table.filters autocomplete should not be visible", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/server_side_filtering_spec_1.ts:124]: Can this be single test please?

**Pull request merged**